--- a/BUILD.gn
+++ b/BUILD.gn
@@ -2598,6 +2598,7 @@ v8_component("v8_libbase") {
 
   if (is_component_build) {
     defines = [ "BUILDING_V8_BASE_SHARED" ]
+    output_extension = "so.6"
   }
 
   if (is_posix) {
@@ -2716,6 +2717,7 @@ v8_component("v8_libplatform") {
 
   if (is_component_build) {
     defines = [ "BUILDING_V8_PLATFORM_SHARED" ]
+    output_extension = "so.6"
   }
 
   public_configs = [ ":libplatform_config" ]
@@ -2900,6 +2902,8 @@ if (is_component_build) {
       "src/v8dll-main.cc",
     ]
 
+    output_extension = "so.6"
+
     public_deps = [
       ":v8_base",
       ":v8_maybe_snapshot",
