diff -Naur a/src/heap/mark-compact.cc b/src/heap/mark-compact.cc
--- a/src/heap/mark-compact.cc	2018-05-18 22:58:23.406934224 -0400
+++ b/src/heap/mark-compact.cc	2018-05-18 22:58:08.596700607 -0400
@@ -3901,6 +3901,8 @@
     }
   }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
   void UpdateTypedPointers() {
     Isolate* isolate = heap_->isolate();
     if (chunk_->typed_slot_set<OLD_TO_NEW, AccessMode::NON_ATOMIC>() !=
@@ -3931,6 +3933,7 @@
           });
     }
   }
+#pragma GCC diagnostic pop
 
   Heap* heap_;
   MarkingState* marking_state_;
