diff --git a/vendor/v8-linux/v8/src/heap/mark-compact.cc b/vendor/v8-linux/v8/src/heap/mark-compact.cc
index 48ae74bc..d532f2f6 100644
--- a/vendor/v8-linux/v8/src/heap/mark-compact.cc
+++ b/vendor/v8-linux/v8/src/heap/mark-compact.cc
@@ -3024,6 +3024,8 @@ class RememberedSetUpdatingItem : public UpdatingItem {
     }
   }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
   void UpdateTypedPointers() {
     if (chunk_->typed_slot_set<OLD_TO_NEW, AccessMode::NON_ATOMIC>() !=
         nullptr) {
@@ -3051,6 +3053,7 @@ class RememberedSetUpdatingItem : public UpdatingItem {
           });
     }
   }
+#pragma GCC diagnostic pop
 
   Heap* heap_;
   MarkingState* marking_state_;
